bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/GoveeHandler.java

```diff
@@ -0,0 +1,394 @@
+    private int lastBrightness;
+    private Color lastColor = new Color(0, 0, 0);
+    private int lastColorTempInKelvin = COLOR_TEMPERATURE_MIN_VALUE.intValue();
+
+    /*
+     * Common Receiver job for the status answers of the devices
+     */
+    public static boolean isRefreshJobRunning() {
+        return refreshJobRunning && THING_HANDLERS.isEmpty();
+    }
+
+    /**
+     * This thing related job <i>thingRefreshSender</i> triggers an update to the Govee device.
+     * The device sends it back to the common port and the response is
+     * then received by the common #refreshStatusReceiver
+     */
+    private final Runnable thingRefreshSender = () -> {
+        try {
+            triggerDeviceStatusRefresh();
+            if (!thing.getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/offline.communication-error.could-not-query-device [\"" + goveeConfiguration.hostname
+                            + "\"]");
+        }
+    };
+
+    public GoveeHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        goveeConfiguration = getConfigAs(GoveeConfiguration.class);
+
+        final String ipAddress = goveeConfiguration.hostname;
+        if (!ipAddress.isEmpty()) {
+            THING_HANDLERS.put(ipAddress, this);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.configuration-error.ip-address.missing");
+            return;
+        }
+        if (!THING_HANDLERS.isEmpty()) {
+            startRefreshStatusJob();
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+        if (triggerStatusJob == null) {
+            logger.debug("Starting refresh trigger job for thing {} ", thing.getLabel());
+
+            triggerStatusJob = scheduler.scheduleWithFixedDelay(thingRefreshSender, 100,
+                    goveeConfiguration.refreshInterval * 1000L, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    /**
+     * Stop the Refresh Status Job, so the same socket can be used for something else (like discovery)
+     */
+    @SuppressWarnings("null")
+    public static void stopRefreshStatusJob() {
+        if (refreshStatusJob != null) {
+            refreshStatusJob.cancel(true);
+        }
+        refreshStatusJob = null;
+        refreshJobRunning = false;
+    }
+
+    /**
+     * (re)start the refresh status job
+     */
+    public static synchronized void startRefreshStatusJob() {
+        if (refreshStatusJob == null) {
+            refreshStatusJob = ThreadPoolManager.getScheduledPool("goveeThingHandler")
+                    .scheduleWithFixedDelay(new RefreshStatusReceiver(), 100, 1000, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    @Override
+    @SuppressWarnings("null")
```

Comment:
Why do you need this?

Reply:
Because without it the build comes up with that warning even though it can never happen and IMHO the warning is actually wrong.

[WARNING] /Users/stefan.hoehn/Development/checkout/openhab-addons/bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/GoveeHandler.java:[174,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] /Users/stefan.hoehn/Development/checkout/openhab-addons/bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/GoveeHandler.java:[195,13] Potential null pointer access: this expression has a '@Nullable' type

see

``` 
173       if (refreshStatusJob != null) {
174            refreshStatusJob.cancel(true);
175       }
```

Reply:
This is because another thread could have set the variable to `null` between the check and the reference. You can fix it like this:
```java
ScheduledFuture<?> refreshStatusJob = this.refreshStatusJob;
if (refreshStatusJob != null) {
    refreshStatusJob.cancel(true);
    this.refreshStatusJob = null;
}
```